python/cog/predictor.py

```diff
@@ -184,15 +199,18 @@ def cleanup(self) -> None:
                     value.unlink()
 
 
-def get_predict(predictor):
-    if hasattr(predictor, "predict"):
-        return predictor.predict
-    return predictor
+def get_run_method(runnable: Runnable):
+    if hasattr(runnable, "predict"):
+        return runnable.predict
+    if hasattr(runnable, "train"):
+        return runnable.train
```

Comment:
We might want be more explicit here about which one we expect on which type:

```suggestion
    if isinstance(runnable, BasePredictor) and hasattr(runnable, "predict"):
        return runnable.predict
    if isinstance(runnable, BaseTrainer) and hasattr(runnable, "train"):
        return runnable.train
```

Reply:
Done, except for backwards compatibility, I kept

```
if hasattr(runnable, "predict"):
    return runnable.predict
```

Since it wasn't enforced that the user extend `BasePredictor`.